docs/docs/starlark-reference/directory.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Directory
+sidebar_label: Directory
+---
+
+The `Directory` constructor creates a `Directory` object that represents a directory inside an existing service (see
+the [ServiceConfig][service-config] object).
+Directory object can be either a files artifact or a persistent directory, depending on the arguments passed to the 
```

Comment:
Can't it actually be both - populated by a files artifact, AND a persistent directory?

Reply:
Nope populating both will throw an interpreation error right now. 
What would you expect to happen in this case? For the files artifact to be mounted to the persistent directory? B/c files artifact are immutable right now, this would basically be equivalent to mounting a regular files artifact.